WordPress/src/main/java/org/wordpress/android/ui/posts/PostListFragment.kt

```diff
@@ -259,9 +260,9 @@ class PostListFragment : Fragment() {
         }
     }
 
-    private fun updatePagedListData(pagedListData: PagedPostList, scrollPosition: Int?) {
+    private fun updatePagedListData(pagedListData: PagedPostList) {
         postListAdapter.submitList(pagedListData)
-        scrollPosition?.let { recyclerView?.smoothScrollToPosition(it) }
+        viewModel.onDataUpdated(pagedListData)
```

Comment:
I am not sure what the benefit of having the fragment do this over just calling this from the `ViewModel` itself. I feel like it exposes information to the Fragment that doesn't necessarily need it and it requires the fragment to have extra logic. Am I maybe missing something?

Reply:
I feel like it's more clear we try to find the post and scroll to it after the adapter is updated... I'm not even sure how to implement it cleanly. The LiveData are updated within the PagedListWrapper and I feel like we don't want to cause side-effects in the ViewModel neither we want to observer the LiveData in the ViewModel. Am I missing something?

Reply:
P.S. We could do it in a "map" or similar. We could also use a MediatorLiveData, but I can't say I'm a fan of either of these solutions. 

Wdyt?

Reply:
After thinking about this, you are absolutely right... it doesn't make any sense to invoke the "onDataUpdated" from the Fragment. I'll use the MediatorLiveData when I get back.

Reply:
Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/9301/commits/8cb2b9601ccbea3719f5088a244af5b5844254f7